core/app/models/spree/promotion/rules/nth_order.rb

```diff
@@ -0,0 +1,43 @@
+module Spree
+  class Promotion
+    module Rules
+      class NthOrder < PromotionRule
+        preference :nth_order, :integer, default: 2
+        # It does not make sense to have this apply to the first order using preferred_nth_order == 1
+        # Instead we could use the first_order rule
```

Comment:
Would it make sense to deprecate the first_order rule and allow this to be `greater_than: 0`?  I see the first_order rule does some special lookup-by-email stuff. Is that an intentional difference?  Also, the first_order rule allows the current order to already be complete (the `&& completed_orders.first != order`). Is there a reason it needs that check but we don't here? (i.e. your `+ 1` below would exclude the current order if it's already complete).